internal/venice-common/src/main/java/com/linkedin/venice/kafka/TopicManager.java

```diff
@@ -77,64 +85,118 @@ public class TopicManager implements Closeable {
   private final String pubSubBootstrapServers;
   private final long kafkaOperationTimeoutMs;
   private final long topicMinLogCompactionLagMs;
-  private final PubSubAdminAdapterFactory<PubSubAdminAdapter> pubSubAdminAdapterFactory;
-  // TODO: Use single PubSubAdminAdapter for both read and write operations
-  private final Lazy<PubSubAdminAdapter> pubSubWriteOnlyAdminAdapter;
-  private final Lazy<PubSubAdminAdapter> pubSubReadOnlyAdminAdapter;
+  private final Lazy<PubSubAdminAdapter> pubSubDefaultAdminAdapter;
+  private final Lazy<PubSubConsumerAdapter> pubSubConsumerAdapter;
+  private final Map<String, PubSubAdminAdapter> pubSubAdminAdapterMap = new HashMap<>();
+  private final Map<String, PubSubConsumerAdapter> pubSubConsumerAdapterMap = new HashMap<>();
   private final PartitionOffsetFetcher partitionOffsetFetcher;
-
+  private final Map<String, PubSubClientsFactory> pubSubClientsFactoryMap;
+  private final PubSubTopicRepository pubSubTopicRepository;
+
+  private final PubSubMessageDeserializer pubSubMessageDeserializer = new PubSubMessageDeserializer(
+      new KafkaValueSerializer(),
+      new LandFillObjectPool<>(KafkaMessageEnvelope::new),
+      new LandFillObjectPool<>(KafkaMessageEnvelope::new));
+  private final Optional<MetricsRepository> optionalMetricsRepository;
+  private final TopicManagerRepository.ClusterNameSupplier clusterNameSupplier;
+  private final TopicManagerRepository.SSLPropertiesSupplier pubSubPropertiesSupplier;
   // It's expensive to grab the topic config over and over again, and it changes infrequently. So we temporarily cache
   // queried configs.
   Cache<PubSubTopic, PubSubTopicConfiguration> topicConfigCache =
       Caffeine.newBuilder().expireAfterWrite(5, TimeUnit.MINUTES).build();
 
+  private PubSubConsumerAdapter getConsumerAdapter(PubSubTopic pubSubTopic) {
+    if (pubSubTopic == null) {
+      return pubSubConsumerAdapter.get();
+    }
+    Optional<String> clusterName = clusterNameSupplier.get(pubSubTopic);
+    if (!clusterName.isPresent()) {
+      return pubSubConsumerAdapter.get();
+    }
+    return pubSubConsumerAdapterMap.computeIfAbsent(
```

Comment:
Is it possible that `getConsumerAdapter` being called concurrently from different threads?  `pubSubConsumerAdapterMap` probably needs to be a `VeniceConcurrentHashMap`, if it is.